JSONModel/JSONModel/JSONModel.m

```diff
@@ -721,6 +718,29 @@ -(void)__inspectProperties
                              );
 }
 
+-(void)__addCustomSetterForName:(NSString*) name
+                     withSuffix:(NSString*) suffix
+                            key:(NSString*) key
+                     toProperty:(JSONModelClassProperty*) p{
+    if (p.customSetters[key])
+        return;
+    
+    SEL setter = NSSelectorFromString([NSString stringWithFormat:@"set%@With%@:", name, suffix]);
```

Comment:
Please move all this logic into `JSONModelCustomSetter` - probably something like:

```objc
JSONModelCustomSetter *setter = [JSONModelCustomSetter setterForObj:self propertyName:name valueType:suffix];

if (setter)
  p.customSetters[key] = setter;
```